src/lib/Microsoft.Health.Events/EventConsumers/Service/Infrastructure/EventPartition.cs

```diff
@@ -54,19 +54,25 @@ public int GetPartitionBatchCount()
             return _partition.Count;
         }
 
-        // flush a fixed number of events
-        public Task<List<IEventMessage>> Flush(int numEvents)
+        public Task<List<IEventMessage>> FlushMaxEvents(int maxEvents)
         {
             var count = 0;
             var events = new List<IEventMessage>();
 
-            while (count < numEvents)
+            while (count < maxEvents)
```

Comment:
Suggestion: Code can be simplified if 
```
while (count < maxEvents && _partition.TryDequeue(out var dequeuedEvent))
```
This should eliminate the need for the if/else with the break